js/src/logger.ts

```diff
@@ -24,6 +24,7 @@ import {
   isEmpty,
   LazyValue,
 } from "./util";
+import { EvalCase } from "./framework";
```

Comment:
yikes, now there's a circular dependency between `logger` and `framework`. I'm surprised javascript doesn't complain about this. Should we factor this out into `util`?

Reply:
I think javascript doesn't complain because it's just a type. The simpler thing to do would be to define it in logger.ts and import it in framework.ts